### PR TITLE
metrics, ddl, server: Add metrics for ddl and server

### PR DESCRIPTION
**Pull request by shenli**
* DDL: is owner
* Server:  events start/stop/kill/hang

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -328,6 +332,8 @@ func (s *Server) ShowProcessList() []util.ProcessInfo {
 func (s *Server) Kill(connectionID uint64, query bool) {
 	s.rwlock.Lock()
 	defer s.rwlock.Unlock()
+	log.Infof("[server] Kill connectionID %d, query %t]", connectionID, query)
+	metrics.ServerEventCounter.WithLabelValues(metrics.EventGracefulDown).Inc()
```
Comment (jackysp) [inline server/server.go]: This should not be `GracefulDown`.

**Comment by shenli** [comment]
@jackysp PTAL

File: ddl/ddl.go [hunk 2]
```diff
@@ -291,7 +291,9 @@ func newDDL(ctx goctx.Context, etcdCli *clientv3.Client, store kv.Storage,
 
 	d.start(ctx)
 	variable.RegisterStatistics(d)
+
 	log.Infof("[ddl] start DDL:%s", d.uuid)
+	metrics.DDLWorkerHistogram.WithLabelValues(metrics.CreateDDLWorker, metrics.RetLabel(nil)).Observe(time.Since(startTime).Seconds())
```
Comment (zimulala) [inline ddl/ddl.go]: Could we rename it to `CreateDDL`?

**Comment by jackysp** [comment]
LGTM

**Comment by shenli** [comment]
@zimulala PTAL

File: metrics/server.go [hunk 3]
```diff
@@ -63,6 +63,21 @@ var (
 			Name:      "critical_error",
 			Help:      "Counter of critical errors.",
 		})
+
+	EventStart        = "server_start"
+	EventGracefulDown = "server_graceful_shutdown"
+	// EventHange occurs when the server.Kill() function is called.
```
Comment (XuHuaiyu) [inline metrics/server.go]: s/EventHange/ EventKill

File: metrics/ddl.go [hunk 4]
```diff
@@ -89,6 +89,17 @@ var (
 			Help:      "Bucketed histogram of processing time (s) of ddl worker operations",
 			Buckets:   prometheus.ExponentialBuckets(0.001, 2, 20),
 		}, []string{"op", "result_state"})
+
+	CreateDDL       = "create_ddl"
+	CreateDDLWorker = "create_ddl_worker"
+	IsDDLOwner      = "is_ddl_worker"
```
Comment (XuHuaiyu) [inline metrics/ddl.go]: s/is_ddl_worker/ is_ddl_owner

File: ddl/ddl_worker.go [hunk 5]
```diff
@@ -85,6 +85,9 @@ func cleanNotify(ch chan struct{}) {
 func (d *ddl) isOwner() bool {
 	isOwner := d.ownerManager.IsOwner()
 	log.Debugf("[ddl] it's the job owner %v, self id %s", isOwner, d.uuid)
+	if isOwner {
+		metrics.DDLCounter.WithLabelValues(metrics.IsDDLOwner).Inc()
```
Comment (XuHuaiyu) [inline ddl/ddl_worker.go]: what does this counter for?
It seems that this function will be called for every ddl sql,

Comment (shenli) [reply ddl/ddl_worker.go]: We could find which tidb-server is owner from grafana. It is easier that use the `admin show ddl` statement.

**Comment by shenli** [comment]
@XuHuaiyu PTAL

**Pull request merged**